src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs

```diff
@@ -88,11 +90,25 @@ public MediaType(string mediaType, int offset, int? length)
             if (subTypeLength == 0)
             {
                 SubType = new StringSegment();
+                SubTypeWithoutSuffix = new StringSegment();
+                SubTypeSuffix = new StringSegment();
                 return;
             }
             else
             {
                 SubType = subType;
+                
+                var subtypeWithoutSuffixLength = TryGetSuffixLength(subType, out var subtypeSuffixLength)
+                    ? subType.Length - subtypeSuffixLength - 1
+                    : subType.Length;
+
+                SubTypeWithoutSuffix = subtypeWithoutSuffixLength > 0
+                    ? subType.Subsegment(0, subtypeWithoutSuffixLength)
+                    : new StringSegment();
+
+                SubTypeSuffix = subtypeSuffixLength > 0
+                    ? subType.Subsegment(subtypeWithoutSuffixLength + 1, subtypeSuffixLength)
+                    : new StringSegment();
```

Comment:
Consider just rolling all of these into a single `if`. This seems like it could be simplified into 'has suffix' and 'trivial' cases